addons/mail/static/src/model/model_core.js

```diff
@@ -2,134 +2,199 @@
 
 /**
  * Module that contains registry for adding new models or patching models.
- * Useful for model manager in order to generate model classes.
+ * Useful for model manager in order to generate models.
  *
  * This code is not in model manager because other JS modules should populate
  * a registry, and it's difficult to ensure availability of the model manager
  * when these JS modules are deployed.
  */
-
-const registry = {};
-
-//------------------------------------------------------------------------------
-// Private
-//------------------------------------------------------------------------------
+export const registry = {};
 
 /**
- * @private
- * @param {string} modelName
- * @returns {Object}
+ * Adds the provided fields to the model specified by the `modelName`.
+ *
+ * @param {string} modelName The name of the model to which to add the fields.
+ * @param {Object} fields Fields to be added.
  */
-function _getEntryFromModelName(modelName) {
-    if (!registry[modelName]) {
-        registry[modelName] = {
-            dependencies: [],
-            factory: undefined,
-            name: modelName,
-            patches: [],
-        };
+export function addFields(modelName, fields) {
+    const definition = registry[modelName];
+    if (!definition) {
+        throw new Error(`Cannot add fields to model "${modelName}": model must be registered before being patched.`);
```

Comment:
It was intentional. I use "patch" here to mean "apply after the original definition". It may be ambiguous, but "added" would sound even more confusing to me